### PR TITLE
Remove multiversion docs

### PR DESCRIPTION
**Pull request by freetonik**

### CODE REVIEW

**Review by rbrazinskas — COMMENTED** [review]
Neither of the four PRs for dropping multiversion documentation support updates ``CHANGELOG``. Was that intentional?

Otherwise LGTM

File: tox.ini [hunk 1]
```diff
@@ -40,21 +40,15 @@ commands =
...
     docs
-allowlist_externals =
-    sphinx-multiversion
 commands =
-    python tox_docs_pre_build_helper.py --build_dir "{env:BUILDDIR}" --build_type "{env:BUILD}"
-    # removed -W to prevent build failure due to
-    # WARNING: don't know which module to import for autodocumenting '__init__' (try placing a "module" or "currentmodule" directive in the document, or giving an explicit module name)
-    sphinx-multiversion "{env:DOCSDIR}" "{env:BUILDDIR}/{env:BUILD}" --dev-name dev  --skip-if-outputdir-exists
-    python tox_docs_post_build_helper.py --build_dir "{env:BUILDDIR}" --build_type "{env:BUILD}"
+    sphinx-build -b {env:BUILD} "{env:DOCSDIR}" "{env:BUILDDIR}/{env:BUILD}"
```
Comment (rbrazinskas) [inline tox.ini:51]: ``cirq-on-iqm``, ``cortex-cli``, and ``qiskit-on-iqm`` use ``-W`` switch. Should we consider adding it here for consistency?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -16,8 +16,6 @@ jobs:
 
     steps:
       - uses: actions/checkout@v3
```
Comment (rbrazinskas) [inline .github/workflows/ci.yml:18]: Just FYI, ``actions/checkout`` and ``actions/setup-python`` have new versions available (``v4`` and ``v5`` respectively). Should we consider upgrading?

File: tox.ini (hunk 1 above)
Comment (freetonik) [reply tox.ini:51]: There's a warning I didn't know how to resolve, and I believe it was the reason why `-W` was originally missing from the config in this repo :D

```
don't know which module to import for autodocumenting '__init__' (try placing a "module" or "currentmodule" directive in the document, or giving an explicit module name)
```

Comment (rbrazinskas) [reply tox.ini:51]: It is a bit confusing as it builds the documents without warnings on my setup. Therefore I thought we could use the ``-W`` switch :-)

Comment (freetonik) [reply tox.ini:51]: Could you reproduce it in this branch? I cleaned everything and re-ran in a new env, still getting the warning.

Also see this older pipeline for example https://github.com/iqm-finland/iqm-client/actions/runs/7883589940/job/21510831653

There are a bunch of `WARNING: don't know which module to import for autodocumenting ...` but the configuration there does not treat warnings as errors, so the pipeline passes.

Comment (freetonik) [reply tox.ini:51]: Ha, interesting... I think we can't stop testing with older Python versions right now, so let's leave the build command as is

File: .github/workflows/ci.yml (hunk 2 above)
Comment (freetonik) [reply .github/workflows/ci.yml]: Updated the PR, let's see how it works

**Review by rbrazinskas — DISMISSED** [review]

**Review by rbrazinskas — APPROVED** [review]

**Pull request merged**